### PR TITLE
(bugfix) Unhandled exception - Abort raised when write to the jaeger-…

### PR DESCRIPTION
**Pull request by ankit-varma10**
…agent fails

Signed-off-by: ankit.varma10 <ankit.varma.ub@gmail.com>

### CODE REVIEW

**Comment by isaachier** [comment]
@ankit-varma10 hello?

File: src/jaegertracing/UDPTransport.cpp [hunk 1]
```diff
@@ -113,7 +113,18 @@ int UDPTransport::flush()
     thrift::Batch batch;
     batch.__set_process(_process);
     batch.__set_spans(_spanBuffer);
-    _client->emitBatch(batch);
+
+    try {
+	    _client->emitBatch(batch);
+    } catch (const std::logic_error &ex) {
```
[the remark this answers is not in the document]
Comment (ankit-varma10) [reply src/jaegertracing/UDPTransport.cpp]: Hey,

Sorry, was caught up in a bunch of other things.

"Why only logic error?"
The only two exceptions I see being raised from UDPClient::emitBatch were logic_error (packet size too big for a single UDP) and std::system_error. So, I think I could add one more handler to it for std:: exception.

    catch (logic) {
    }
    catch (system) {

    } catch (std:exception) {

    }

Thoughts?

[the remark this answers is not in the document]
Comment (isaachier) [reply src/jaegertracing/UDPTransport.cpp]: OK cool. My idea what just skip the `catch (logic) {}` and do the rest as above.

[the remark this answers is not in the document]
Comment (ankit-varma10) [reply src/jaegertracing/UDPTransport.cpp]: Oh yeah. I missed that.

Makes sense. I'll make the changes today.

**Comment by isaachier** [comment]
Awesome. I see there is no coverage for this new code. If you don't want to add a test case, I can do it. Just let me know so I can proceed if necessary. Thanks!

**Comment by isaachier** [comment]
Just an interesting follow up. @manannayak pointed out to me that the `RemoteReporter::sweepQueue` method should have caught the exception instead of terminating the application. I believed he was correct until I found that `RemoteReporter::flush` was declared `noexcept` and throwing, thus violating all guarantees of further exception handling up the stack. Glad someone caught this. Might be worth adding a `catch (...)` block in `RemoteReporter::flush` to avoid issues like this in the future.

**Pull request merged**